packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -178,7 +179,7 @@ module.exports = {
       // First, run the linter.
       // It's important to do this before Babel processes the JS.
       {
-        test: /\.(js|mjs|jsx)$/,
+        test: /\.(js|mjs|jsx|ts|tsx)$/,
```

Comment:
Should we enable eslint for typescript files? Would we need to use `typescript-eslint-parser`? Should we bring tslint support back?

Reply:
`typescript-eslint-parser` was pre-babel support afaik, since babel can now parse typescript and our eslint setup uses the babel parser, this is probably fine.

Reply:
AFAIK eslint is fine for typescript now that babel supports it.

Reply:
Im on the fence for this one. 

On the one hand it may be confusing for those of us that would like to use TSLint for rules that are available when we have types, such as https://palantir.github.io/tslint/rules/no-floating-promises/. Do we run TSLint separately as well?

On the other hand we keep the same linting rules as vanilla CRA which is a great starting point for most people.

Reply:
We will take the same stance with TSLint as we do ESLint (for now).
You will not be permitted to enable any rules that enforce style/coding practice. The use of ESLint is strictly to find errors -- something like TSLint seems redundant because the point of TypeScript is to "find errors", haha.

You can always run TSLint yourself like people do ESLint now and have an editor integration or separate lint step on commit.